### PR TITLE
supporting vSphere IPI cleanup

### PR DESCRIPTION
**Pull request by vavuthu**
Signed-off-by: vavuthu <vavuthu@redhat.com>

### CODE REVIEW

**Comment by vavuthu** [comment]
vsphere IPI cluster

```
$ vsphere-cleanup --cluster-name vavuthu-ipi1 --ocsci-conf ~/VJ/VMWARE/UPLOAD/new/new/vSphere-DC-ECO-new_VC1 
2022-10-27 11:11:33,542 ocs_ci.cleanup.vsphere.cleanup [INFO] Resource pool vavuthu-ipi1 does not exists
2022-10-27 11:11:33,542 ocs_ci.cleanup.vsphere.cleanup [INFO] Deleting vSphere IPI nodes
2022-10-27 11:13:08,138 ocs_ci.cleanup.vsphere.cleanup [INFO] vavuthu-ipi1-kt5ws-worker-sxhmx
2022-10-27 11:13:08,582 ocs_ci.cleanup.vsphere.cleanup [INFO] vavuthu-ipi1-kt5ws-worker-wwx79
2022-10-27 11:13:09,043 ocs_ci.cleanup.vsphere.cleanup [INFO] vavuthu-ipi1-kt5ws-worker-wvthz
2022-10-27 11:13:09,490 ocs_ci.cleanup.vsphere.cleanup [INFO] vavuthu-ipi1-kt5ws-master-2
2022-10-27 11:13:09,934 ocs_ci.cleanup.vsphere.cleanup [INFO] vavuthu-ipi1-kt5ws-master-0
2022-10-27 11:13:10,379 ocs_ci.cleanup.vsphere.cleanup [INFO] vavuthu-ipi1-kt5ws-master-1
2022-10-27 11:13:10,831 ocs_ci.cleanup.vsphere.cleanup [INFO] vavuthu-ipi1-kt5ws-rhcos
2022-10-27 11:13:17,802 ocs_ci.utility.vsphere [INFO] power state of vavuthu-ipi1-kt5ws-worker-sxhmx: poweredOn
2022-10-27 11:13:18,254 ocs_ci.utility.vsphere [INFO] power state of vavuthu-ipi1-kt5ws-worker-wwx79: poweredOn
2022-10-27 11:13:19,185 ocs_ci.utility.vsphere [INFO] power state of vavuthu-ipi1-kt5ws-worker-wvthz: poweredOn
2022-10-27 11:13:20,087 ocs_ci.utility.vsphere [INFO] power state of vavuthu-ipi1-kt5ws-master-2: poweredOn
2022-10-27 11:13:20,907 ocs_ci.utility.vsphere [INFO] power state of vavuthu-ipi1-kt5ws-master-0: poweredOn
2022-10-27 11:13:21,726 ocs_ci.utility.vsphere [INFO] power state of vavuthu-ipi1-kt5ws-master-1: poweredOn
2022-10-27 11:13:22,409 ocs_ci.utility.vsphere [INFO] power state of vavuthu-ipi1-kt5ws-rhcos: poweredOff
2022-10-27 11:13:24,644 ocs_ci.utility.vsphere [INFO] Powering off VMs: ['vavuthu-ipi1-kt5ws-worker-sxhmx', 'vavuthu-ipi1-kt5ws-worker-wwx79', 'vavuthu-ipi1-kt5ws-worker-wvthz', 'vavuthu-ipi1-kt5ws-master-2', 'vavuthu-ipi1-kt5ws-master-0', 'vavuthu-ipi1-kt5ws-master-1']
2022-10-27 11:13:31,271 ocs_ci.utility.vsphere [INFO] Destroying VM's: ['vavuthu-ipi1-kt5ws-worker-sxhmx', 'vavuthu-ipi1-kt5ws-worker-wwx79', 'vavuthu-ipi1-kt5ws-worker-wvthz', 'vavuthu-ipi1-kt5ws-master-2', 'vavuthu-ipi1-kt5ws-master-0', 'vavuthu-ipi1-kt5ws-master-1', 'vavuthu-ipi1-kt5ws-rhcos']
2022-10-27 11:13:36,186 ocs_ci.cleanup.vsphere.cleanup [INFO] Removing IP for node vavuthu-ipi1-0 from IPAM server
2022-10-27 11:13:36,296 ocs_ci.cleanup.vsphere.cleanup [INFO] Removing IP for node vavuthu-ipi1-1 from IPAM server
2022-10-27 11:13:36,402 ocs_ci.deployment.vmware [INFO] Deleting DNS records
2022-10-27 11:13:36,410 botocore.credentials [INFO] Found credentials in shared credentials file: ~/.aws/credentials
2022-10-27 11:13:47,807 ocs_ci.deployment.vmware [INFO] hosted zone id: /hostedzone/ZXXX
2022-10-27 11:13:47,807 /home/vavuthu/VJ/projects/ocs-ci/ocs_ci/utility/aws.py [INFO] deleting record set: api.vavuthu-ipi1.qe.rh-ocs.com.
2022-10-27 11:13:48,268 /home/vavuthu/VJ/projects/ocs-ci/ocs_ci/utility/aws.py [INFO] deleting record set: \052.apps.vavuthu-ipi1.qe.rh-ocs.com.

```

**Review by petr-balogh — APPROVED** [review]

**Review by AviadP — APPROVED** [review]

File: ocs_ci/cleanup/vsphere/cleanup.py [hunk 1]
```diff
@@ -123,14 +149,37 @@ def delete_ips(self, cluster_name):
...
+        Args:
+            cluster_name (str): Name of the cluster to release IP's
+                from IPAM server
+
+        """
+        for i in range(0, 2):
+            node = f"{cluster_name}-{i}"
+            self.delete(node)
+
+    def delete(self, node):
```
Comment (ebenahar) [inline ocs_ci/cleanup/vsphere/cleanup.py:168]:
```suggestion
    def delete_ips(self, node):
```

File: ocs_ci/cleanup/vsphere/cleanup.py [hunk 2]
```diff
@@ -123,14 +149,37 @@ def delete_ips(self, cluster_name):
...
+    def delete_ips_ipi(self, cluster_name):
+        """
+        Delete IP's from IPAM server for IPI cluster
+
+        Args:
+            cluster_name (str): Name of the cluster to release IP's
+                from IPAM server
+
+        """
+        for i in range(0, 2):
```
Comment (ebenahar) [inline ocs_ci/cleanup/vsphere/cleanup.py:164]: can there be more than 3 IPs to delete? What if there are more than 3 nodes?

Comment (vavuthu) [reply ocs_ci/cleanup/vsphere/cleanup.py:164]: those are the API and ingress IP's ( not the node IP's ). Its always 2 IP's

Comment (ebenahar) [reply ocs_ci/cleanup/vsphere/cleanup.py:164]: ack

**Review by ebenahar — APPROVED** [review]

**Pull request merged**